loadtest/control/gencontroller/state.go

```diff
@@ -112,6 +122,35 @@ func (st *state) storeChannelID(channelID string) {
 	st.channels = append(st.channels, channelID)
 }
 
+func (st *state) storeUserID(userID string) {
+	st.usersMut.Lock()
+	defer st.usersMut.Unlock()
+	st.users = append(st.users, userID)
+}
+
+func (st *state) numUsers() int {
+	st.usersMut.RLock()
+	defer st.usersMut.RUnlock()
+	return len(st.users)
+}
+
+func (st *state) genCombinations() {
+	st.userCombinationsOnce.Do(func() {
+		st.userCombinations = combinations.Combinations(st.users, 2)
+	})
+}
+
+// getUserPair returns the pair at the current index
+// and increments the index
+func (st *state) getUserPair() []string {
+	st.usersMut.Lock()
+	defer st.usersMut.Unlock()
+
+	pair := st.userCombinations[st.userCombinationsIndex]
+	st.userCombinationsIndex++
+	return pair
+}
```

Comment:
Maybe we should check the array length here

Reply:
Indeed, I got a panic here. I don't know why it was empty, though, maybe it's some issue with my config.
```
panic: runtime error: index out of range [0] with length 0

goroutine 182 [running]:
github.com/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller.(*state).getUserPair(0xc0001e2460)
	/home/alejandro/projects/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller/state.go:149 +0xf5
github.com/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller.(*GenController).createDirectChannel(0xc0003ce540, {0x15248e8, 0xc001dfd800})
	/home/alejandro/projects/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller/actions.go:136 +0x215
github.com/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller.(*GenController).runActions(0xc0003ce540, 0xc002793b20?, 0xc002b8fa68)
	/home/alejandro/projects/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller/controller.go:232 +0x78
github.com/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller.(*GenController).Run(0xc0003ce540)
	/home/alejandro/projects/mattermost/mattermost-load-test-ng/loadtest/control/gencontroller/controller.go:160 +0xa66
github.com/mattermost/mattermost-load-test-ng/loadtest.(*LoadTester).addUser.func1()
	/home/alejandro/projects/mattermost/mattermost-load-test-ng/loadtest/loadtest.go:131 +0x22
created by github.com/mattermost/mattermost-load-test-ng/loadtest.(*LoadTester).addUser
	/home/alejandro/projects/mattermost/mattermost-load-test-ng/loadtest/loadtest.go:130 +0x38b
exit status 2
```